### ISSUE TITLE
load the yolov8 model

### ISSUE THREAD

**Issue by nip6** [ultralytics/ultralytics#2176]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

The above is the method of importing v7 and v5, and the following is the method of v8, but it seems that the results obtained by both methods when importing the model are inconsistent. Can I combine this as I did before?

how can I do it?

### Additional

![image](https://user-images.githubusercontent.com/107199746/233671179-3923f068-b4ea-4288-8eda-425752ace436.png)

Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\envs\YDS\lib\site-packages\IPython\core\interactiveshell.py", line 3505, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-9531733304c2>", line 1, in <module>
    runfile('C:/Users/14621/Desktop/Y8DS/yolov5-slowfast-deepsort-PytorchVideo/yolo_slowfast.py', wdir='C:/Users/14621/Desktop/Y8DS/yolov5-slowfast-deepsort-PytorchVideo')
  File "D:\pycharm2020\PyCharm 2020.1\plugins\python\helpers\pydev\_pydev_bundle\pydev_umd.py", line 197, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "D:\pycharm2020\PyCharm 2020.1\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "C:/Users/14621/Desktop/Y8DS/yolov5-slowfast-deepsort-PytorchVideo/yolo_slowfast.py", line 170, in <module>
    main(config)
  File "C:/Users/14621/Desktop/Y8DS/yolov5-slowfast-deepsort-PytorchVideo/yolo_slowfast.py", line 121, in main
    yolo_preds.files = [f"img_{i * 25 + k}.jpg" for k in range(img_num)]
AttributeError: 'list' object has no attribute 'files'

**Comment by glenn-jocher** [comment]
@nip6, about your question, firstly, please be aware that this is the support thread for YOLOv8 and we do not discuss YOLOv5 in here.

Regarding your model importing question, each version of YOLO has its own set of weights, and generally you cannot mix the models or weights of one with the other. If we stick to YOLOv8, as you mentioned, the code for importing the model you post here looks correct. However, it's hard to know the cause of the inconsistency without more detailed information regarding input and output. It's possible that the weights of these two models are trained for different datasets, which could result in different results.

About your additional issue, the error message you're seeing seems to be related with the following assignment line:
```python
yolo_preds.files = [f"img_{i * 25 + k}.jpg" for k in range(img_num)]
```
The error message suggests it is because yolo_preds is a list type rather than a YOLO class instance. It's hard to know more based on your information, but double-checking that you're using the correct type and that it is properly initialized could help solve the issue.

**Comment by nip6** [comment]
> 
First of all, thank you very much for your response. Secondly, I would like to elaborate on my question in more detail and see if you can help me solve it.

![image](https://user-images.githubusercontent.com/107199746/233764446-2d69f103-7bbf-4d50-b3d7-cb6a7529a4b3.png)

This is a debug scenario when I was trying to import YOLOv5 using the command shown in the image. From here, it can be seen that the variable 'yolo_preds' is not yet a list at this point.

![image](https://user-images.githubusercontent.com/107199746/233764951-93f6d660-6df4-437a-b1aa-53cc3e7353b6.png)

This is a debug scenario in which I imported the v8 model using the command shown in the image. From the image, it can be seen that the variable 'yolo_preds' has already become a list at this point.

So I would like to ask if you have any methods to import the YOLOv8 model into my program like the YOLOv5 model without making significant changes to the code, and have it run normally as usual. I would be very grateful if I could receive your help.

**Comment by glenn-jocher** [comment]
@nip6 thank you for providing more details about your question. Based on your description, it seems like the two models, YOLOv5 and YOLOv8, have different implementations and are not interchangeable. Therefore, it's necessary to make changes to the code to switch from one model to another.

However, the good news is that the usage of YOLOv8 is very similar to that of YOLOv5, so the changes might not be significant. You can still use the same YOLO class to detect objects, but you will need to make sure that you are feeding the correct model and weights to the class.

In summary, to use YOLOv8 in your program, you will need to make sure that you are using the correct model and weights and modify the code to correspond with the new model.

**Comment by nip6** [comment]
> @nip6 thank you for providing more details about your question. Based on your description, it seems like the two models, YOLOv5 and YOLOv8, have different implementations and are not interchangeable. Therefore, it's necessary to make changes to the code to switch from one model to another.
> 
> However, the good news is that the usage of YOLOv8 is very similar to that of YOLOv5, so the changes might not be significant. You can still use the same YOLO class to detect objects, but you will need to make sure that you are feeding the correct model and weights to the class.
> 
> In summary, to use YOLOv8 in your program, you will need to make sure that you are using the correct model and weights and modify the code to correspond with the new model.

Thank you for your response. I am wondering if there is a way to make the model imported using the command 'model = YOLO('yolov8.pt')' the same type as the model imported using the command 'model = torch.hub.load('...', '...')'. Because if they are not the same type, many attributes of the 'model' variable may be missing, which could require significant code changes.

My programming knowledge is limited, and it is difficult for me to perform complicated modification tasks. I would be extremely grateful if you could help me answer this question.
